### ISSUE TITLE
[Enhancement] use `*` in stub signatures to enforce use of keyword arguments

### ISSUE THREAD

**Issue by corleyma** [vemel/mypy_boto3_builder#80]
Many boto3 methods take keyword arguments only, but the stubs are currently generated with all arguments as regular positional arguments.  Adding `*` as the first argument in the function signature would signal to myy/pyright that no positional arguments are expected, and hint to folks that they must supply their arguments as keywords.

E.g.:

```python
class SQSClient:
    def purge_queue(self, *, QueueUrl: str) -> None:
        """
        [Show boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/1.17.79/reference/services/sqs.html#SQS.Client.purge_queue)
        [Show boto3-stubs documentation](./client.md#purge_queue)
        """

**Comment by vemel** [comment]
This is a great idea! I am not sure if it will work on `py36` though, which is still supported. I will take a look if I can make it compatible.

Thank you for the suggestion!

**Comment by corleyma** [comment]
This functionality exists in at least Python 3.5: https://docs.python.org/3.5/reference/compound_stmts.html#function-definitions
> Parameters after “*” or “*identifier” are keyword-only parameters and may only be passed used keyword arguments.

A for Python 3.6, at my company we have many 3.6 projects using this idiom, so should be good to go.

**Comment by vemel** [comment]
I have just added this feature, which should be included in `boto3-stubs 1.17.79` release. Please let me know if it works for you as expected.

**Comment by corleyma** [comment]
Just testing this out.  It looks good, except that it looks like some kwarg-only methods are missed.  Taking SQS as an example:

```python
    def delete_message(self, QueueUrl: str, ReceiptHandle: str) -> None:
        """
        [Show boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/1.17.91/reference/services/sqs.html#SQS.Client.delete_message)
        [Show boto3-stubs documentation](https://vemel.github.io/boto3_stubs_docs/mypy_boto3_sqs/client.html#delete_message)
        """

    def delete_message_batch(
        self, QueueUrl: str, Entries: List[DeleteMessageBatchRequestEntryTypeDef]
    ) -> DeleteMessageBatchResultTypeDef:
        """
        [Show boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/1.17.91/reference/services/sqs.html#SQS.Client.delete_message_batch)
        [Show boto3-stubs documentation](https://vemel.github.io/boto3_stubs_docs/mypy_boto3_sqs/client.html#delete_message_batch)
        """
```

`delete_message` and `delete_message_batch` are also kwargs only (see https://boto3.amazonaws.com/v1/documentation/api/1.17.91/reference/services/sqs.html#SQS.Client.delete_message) but somehow don't get the `*` treatment.  There are other methods in the SQSClient that are also missing it (purge_queue, etc).

**Comment by corleyma** [comment]
By the way, thanks so much for quickly getting a release out though.  It's already a big improvement for the methods that were covered; just need to figure out how to fill the gaps I reckon.

**Comment by vemel** [comment]
Yes, I decided first to make methods with just one argument to allow positional args. Then, after looking at boto3 usage in various projects, I also added positional args to methods with 2 arguments, because people often use them without kwargs.

You can find a history of changes in releases because this is not documented yet.

Please let me know if you see how to improve the implementation.

**Comment by vemel** [comment]
I added `Keyword-only arguments` text to generated docs to somehow document current behavior. E.g.: https://vemel.github.io/boto3_stubs_docs/mypy_boto3_s3/client.html#abort_multipart_upload

**Comment by corleyma** [comment]
So, the problem is that for methods like `delete_message`, it's not possible for people to be using that with positional arguments: boto3 does not accept positional arguments for those methods.  Anything in the boto3 docs with a signature of (**kwargs) can't be given positional arguments, but the boto3-stubs translate those methods into stubs with a bunch of positional args so folks using editor autocomplete, etc, will assume that the code they have written is correct, but it will actually fail at runtime.  That's why I requested adding `*` as the first non-self argument in those method stubs, since now mypy/pyright (and thus people's IDEs) will now understand that the arguments must be specified as kwargs.

In other words, it's not about how _many_ arguments a method in boto3 takes, only about whether the underlying implementation only takes kwargs.

**Comment by vemel** [comment]
As I see, most of the methods use `**kwargs`. I will test if all 1- and 2- argument methods are keyword-only or not. If you can help me with the check - that would be great.

**Comment by vemel** [comment]
All generated methods arguments are now keyword-only. I have just released `boto3-stubs 1.17.92.post1` and regenerated docs. Please let me know if it works as expected.
